ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/AgentUtils.java

```diff
@@ -320,7 +320,15 @@ public static Map<String, String> parseLLMOutput(
             parseThoughtResponse(modelOutput, thoughtResponse);
         } else if (parameters.containsKey(TOOL_CALLS_PATH)) {
             modelOutput.put(THOUGHT_RESPONSE, StringUtils.toJson(dataAsMap));
-            Object response = JsonPath.read(dataAsMap, parameters.get(LLM_RESPONSE_FILTER));
+            Object response;
+            boolean isToolUseResponse = false;
+            try {
+                response = JsonPath.read(dataAsMap, parameters.get(LLM_RESPONSE_FILTER));
```

Comment:
is it easier just to check if LLM_RESPONSE_FILTER in parameters.keys then get the response, else you read from the TOOL_CALLS_PATH? this exception is not necessary

Reply:
```
boolean isToolUseResponse = !parameters.containsKey(LLM_RESPONSE_FILTER);

if (!isToolUseResponse) {
    response = JsonPath.read(dataAsMap, parameters.get(LLM_RESPONSE_FILTER));
} else {
    response = JsonPath.read(dataAsMap, parameters.get(TOOL_CALLS_PATH));
}
```